Intermittent failure in p2p_1p1c_network.py", line 58, in raise_network_minfee                                        assert_greater_than(node.getmempoolinfo()['mempoolminfee'], FEERATE_1SAT_VB)     ; AssertionError: 0.00001000 <= 0.00001000

https://github.com/bitcoin/bitcoin/actions/runs/10923649833/job/30320807014?pr=30921#step:7:46110

```
node0 2024-09-18T14:10:27.733922Z [httpworker.2] [src/rpc/request.cpp:232] [parse] [rpc] ThreadRPCServer method=getmempoolinfo user=__cookie__ 
 test  2024-09-18T14:10:27.734000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/Users/runner/work/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 132, in main
                                       self.run_test()
                                     File "/Users/runner/work/bitcoin/bitcoin/ci/scratch/build-aarch64-apple-darwin23.6.0/test/functional/p2p_1p1c_network.py", line 137, in run_test
                                       self.raise_network_minfee()
                                     File "/Users/runner/work/bitcoin/bitcoin/ci/scratch/build-aarch64-apple-darwin23.6.0/test/functional/p2p_1p1c_network.py", line 58, in raise_network_minfee
                                       assert_greater_than(node.getmempoolinfo()['mempoolminfee'], FEERATE_1SAT_VB)
                                     File "/Users/runner/work/bitcoin/bitcoin/test/functional/test_framework/util.py", line 82, in assert_greater_than
                                       raise AssertionError("%s <= %s" % (str(thing1), str(thing2)))
                                   AssertionError: 0.00001000 <= 0.00001000

Encountered this in PR #30946 as well.

```
test  2024-09-22T19:16:03.915000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/Users/runner/work/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 132, in main
                                       self.run_test()
                                     File "/Users/runner/work/bitcoin/bitcoin/ci/scratch/build-aarch64-apple-darwin23.6.0/test/functional/p2p_1p1c_network.py", line 137, in run_test
                                       self.raise_network_minfee()
                                     File "/Users/runner/work/bitcoin/bitcoin/ci/scratch/build-aarch64-apple-darwin23.6.0/test/functional/p2p_1p1c_network.py", line 58, in raise_network_minfee
                                       assert_greater_than(node.getmempoolinfo()['mempoolminfee'], FEERATE_1SAT_VB)
                                     File "/Users/runner/work/bitcoin/bitcoin/test/functional/test_framework/util.py", line 82, in assert_greater_than
                                       raise AssertionError("%s <= %s" % (str(thing1), str(thing2)))
                                   AssertionError: 0.00001000 <= 0.00001000
```

--- a/test/functional/mempool_package_rbf.py
+++ b/test/functional/mempool_package_rbf.py
@@ -554,7 +554,7 @@ class PackageRBFTest(BitcoinTestFramework):
         self.generate(node, 1)
 
     def test_child_conflicts_parent_mempool_ancestor(self):
-        fill_mempool(self, self.nodes[0])
+        fill_mempool(self, self.nodes[0], tx_sync_fun=self.no_op)
         # Reset coins since we filled the mempool with current coins
         self.coins = self.wallet.get_utxos(mark_as_spent=False, confirmed_only=True)
 

--- a/test/functional/p2p_1p1c_network.py
+++ b/test/functional/p2p_1p1c_network.py
@@ -49,9 +49,6 @@ class PackageRelayTest(BitcoinTestFramework):
     def raise_network_minfee(self):
         fill_mempool(self, self.nodes[0])
 
-        self.log.debug("Wait for the network to sync mempools")
-        self.sync_mempools()
-
         self.log.debug("Check that all nodes' mempool minimum feerates are above min relay feerate")
         for node in self.nodes:
             assert_equal(node.getmempoolinfo()['minrelaytxfee'], FEERATE_1SAT_VB)

--- a/test/functional/p2p_tx_download.py
+++ b/test/functional/p2p_tx_download.py
@@ -250,7 +250,7 @@ class TxDownloadTest(BitcoinTestFramework):
     def test_rejects_filter_reset(self):
         self.log.info('Check that rejected tx is not requested again')
         node = self.nodes[0]
-        fill_mempool(self, node)
+        fill_mempool(self, node, tx_sync_fun=self.no_op)
         self.wallet.rescan_utxos()
         mempoolminfee = node.getmempoolinfo()['mempoolminfee']
         peer = node.add_p2p_connection(TestP2PConn())

--- a/test/functional/test_framework/mempool_util.py
+++ b/test/functional/test_framework/mempool_util.py
@@ -19,14 +19,11 @@ from .wallet import (
 )
 
 
-def fill_mempool(test_framework, node):
+def fill_mempool(test_framework, node, *, tx_sync_fun=None):
     """Fill mempool until eviction.
 
     Allows for simpler testing of scenarios with floating mempoolminfee > minrelay
-    Requires -datacarriersize=100000 and
-   -maxmempool=5.
-    It will not ensure mempools become synced as it
-    is based on a single node and assumes -minrelaytxfee
+    Requires -datacarriersize=100000 and -maxmempool=5 and assumes -minrelaytxfee
     is 1 sat/vbyte.
     To avoid unintentional tx dependencies, the mempool filling txs are created with a
     tagged ephemeral miniwallet instance.
@@ -69,9 +66,13 @@ def fill_mempool(test_framework, node):
     batch_fees = [(i + 1) * base_fee for i in range(num_of_batches)]
 
     test_framework.log.debug("Fill up the mempool with txs with higher fee rate")
-    with node.assert_debug_log(["rolling minimum fee bumped"]):
-        for fee in batch_fees:
-            send_batch(fee)
+    for fee in batch_fees[:-3]:
+        send_batch(fee)
+    tx_sync_fun() if tx_sync_fun else test_framework.sync_mempools()  # sync before any eviction
+    assert_equal(node.getmempoolinfo()["mempoolminfee"], Decimal("0.00001000"))
+    for fee in batch_fees[-3:]:
+        send_batch(fee)
+    tx_sync_fun() if tx_sync_fun else test_framework.sync_mempools()  # sync after all evictions
 
     test_framework.log.debug("The tx should be evicted by now")
     # The number of transactions created should be greater than the ones present in the mempool

--- a/test/functional/test_framework/mempool_util.py
+++ b/test/functional/test_framework/mempool_util.py
@@ -57,18 +57,21 @@ def fill_mempool(test_framework, node):
     tx_to_be_evicted_id = ephemeral_miniwallet.send_self_transfer(
         from_node=node, utxo_to_spend=confirmed_utxos.pop(0), fee_rate=relayfee)["txid"]
 
+    def send_batch(fee):
+        utxos = confirmed_utxos[:tx_batch_size]
+        create_lots_of_big_transactions(ephemeral_miniwallet, node, fee, tx_batch_size, txouts, utxos)
+        del confirmed_utxos[:tx_batch_size]
+
     # Increase the tx fee rate to give the subsequent transactions a higher priority in the mempool
     # The tx has an approx. vsize of 65k, i.e. multiplying the previous fee rate (in sats/kvB)
     # by 130 should result in a fee that corresponds to 2x of that fee rate
     base_fee = relayfee * 130
+    batch_fees = [(i + 1) * base_fee for i in range(num_of_batches)]
 
     test_framework.log.debug("Fill up the mempool with txs with higher fee rate")
     with node.assert_debug_log(["rolling minimum fee bumped"]):
-        for batch_of_txid in range(num_of_batches):
-            fee = (batch_of_txid + 1) * base_fee
-            utxos = confirmed_utxos[:tx_batch_size]
-            create_lots_of_big_transactions(ephemeral_miniwallet, node, fee, tx_batch_size, txouts, utxos)
-            del confirmed_utxos[:tx_batch_size]
+        for fee in batch_fees:
+            send_batch(fee)
 
     test_framework.log.debug("The tx should be evicted by now")
     # The number of transactions created should be greater than the ones present in the mempool

